Docs: Cleanup meta field docs

rjernst opened the pull request:
Meta fields were locked down to not allow exotic options to the
underlying field types in #8143. This change fixes the docs
to no longer refer to the old settings.

closes #10879

jpountz commented on docs/reference/mapping/fields/type-field.asciidoc:
```diff
@@ -1,31 +1,10 @@
-    }
-}
---------------------------------------------------
+Note, even though the `_type` is not indexed, all the APIs still work
```
Hmm, but it IS indexed, right?

rjernst replied:
Yes! This was a mistake, I reworded the `_type` docs.

clintongormley commented on docs/reference/mapping/fields/field-names-field.asciidoc:
```diff
@@ -1,9 +1,7 @@
 [[mapping-field-names-field]]
 === `_field_names`
+added[1.3.0]
```
No need for annotations from 1.x

clintongormley commented on docs/reference/mapping/fields/routing-field.asciidoc:
```diff
@@ -5,15 +5,6 @@ The routing field allows to control the `_routing` aspect when indexing
-reindexing data will be possible if the routing value is completely
-external and not part of the docs.
-
-[float]
```
In the next para, delete "or derived from the doc".  

Also, need to check with @martijnvg, but I think you can delete "A delete operation will be broadcasted to all shards if no routing value is provided and `_routing` is required."

rjernst replied:
Looks like this was removed in #10136. I've removed that delete operation line as well.

clintongormley commented on docs/reference/query-dsl/filters/ids-filter.asciidoc:
```diff
@@ -2,9 +2,7 @@
-does not require the <<mapping-id-field,_id>>
-field to be indexed since it works using the
-<<mapping-uid-field,_uid>> field.
+uses the <<mapping-uid-field,_uid>> field.
```
Add "If no `type` is specified, all types defined in the index mapping are tried." (or something like that)

clintongormley commented on docs/reference/query-dsl/filters/type-filter.asciidoc:
```diff
@@ -2,8 +2,7 @@
 Filters documents matching the provided document / mapping type. Note,
-this filter can work even when the `_type` field is not indexed (using
-the <<mapping-uid-field,_uid>> field).
+this filter uses the <<mapping-uid-field,_uid>> field.
```
Types are indexed, no? So surely it doesn't use the UID field?

rjernst replied:
This was just a goof on my part, removed.

clintongormley commented on docs/reference/query-dsl/queries/ids-query.asciidoc:
```diff
@@ -2,9 +2,7 @@
-does not require the <<mapping-id-field,_id>>
-field to be indexed since it works using the
-<<mapping-uid-field,_uid>> field.
+uses the <<mapping-uid-field,_uid>> field.
```
as per comment on the ids filter

rjernst replied:
`_id` is not stored or indexed. The ids query does use `_uid`.

clintongormley commented:
As a side note, shouldn't we remove the ability to configure store/index on timestamp? http://www.elastic.co/guide/en/elasticsearch/reference/master/mapping-timestamp-field.html#_store_index_2

clintongormley commented on docs/reference/mapping/fields/routing-field.asciidoc:
```diff
@@ -5,15 +5,6 @@ The routing field allows to control the `_routing` aspect when indexing
 data and explicit routing control is required.
```
Add a note saying that `_routing` is stored but not indexed.

clintongormley replied:
Actually looks like it is stored AND indexed

rjernst commented:
@clintongormley I pushed a new commit addressing your comments. For `_timestamp` we never locked it down because it is essentially a user field (the only way to add a NOW to a date field). I think we should continue looking at changing that separately.
